src/ApiClient.php

```diff
@@ -114,24 +117,17 @@ private function isResponseHtml($string)
         return ($string != strip_tags($string));
     }
 
-    /**
-     * If (and only if) the entire $in parameter is enclosed in single or double quotes, strip them
-     *
-     * @param string $in
-     * @return string
-     **/
-    public function deQuote($in)
+    private function setInstance($url)
     {
-        $_first = substr($in, 0, 1);
-        $_last = substr($in, -1, 1);
-        if ($_first == $_last && ($_first == '"' || $_first == '\'')) {
-            $in = substr($in, 1, strlen($in) - 2);
+        $this->instance = $url;
+        if (substr($this->instance, -1) === '/') {
```

Comment:
You could replace this whole function with:

    $this->instance = ltrim(rtrim($url, " /'\""), " '\"");

ltrim and rtrim remove spaces and other provided characters from beginning/end of the string. We went to remove any quotes and spaces from both ends, and also any / at the end.